src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2StreamWindowManager.cs

```diff
@@ -0,0 +1,257 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal sealed partial class Http2Connection
+    {
+        private struct Http2StreamWindowManager
+        {
+            public const int StreamWindowUpdateRatio = 8;
+            private static readonly double StopWatchToTimesSpan = TimeSpan.TicksPerSecond / (double)Stopwatch.Frequency;
+
+            private readonly Http2Connection _connection;
```

Comment:
This struct is not very small, so I've been thinking whether we could adjust the code to get rid some of the fields.
E.g.: could we return `windowUpdateIncrement` value from the `AdjustWindow` function (null or 0 or -1 for no update necessary, whatever seems the best) and do the frame sending in H2Stream? Then we wouldn't need neither `_connection` nor `_stream` here.

Reply:
`_connection` is also used to access `_rttEstimator`, and `_stream` has multiple purposes too.

Of course we could flatten all the logic into `Http2Stream` to save a few bytes, but I very much dislike that idea, I prefer maintaining clear separation of concerns over practically unnoticeable performance wins.

Reply:
We can get the connection from the stream, so it seems like we don't need to store the connection.

If we want to avoid storing the Http2Stream too, we can pass it in to methods as necessary, since it's always called from Http2Stream.